compiler-rt/lib/mpk_untrusted/alloc_site_handler.h

```diff
@@ -104,6 +113,10 @@ class AllocSiteHandler {
   std::set<AllocSite> fault_set;
   // Thread safety mutex
   std::mutex mx;
+  // Mapping of thread-id to saved pkey
+  std::map<pid_t, PKeyInfo> pid_key_map;
```

Comment:
Might make sense to name this something like `pkey_by_pid_map`, and then associated functions can be things like `get_pkey_by_pid(pid)`.  

The other alternative is to use a structure to track the metadata instead e.g., PKeyMetadata or whatever.

Note: I have no preference on snake vs camel case, its just what I typed, so use whatever you'd like

Reply:
Changed to suggested name pkey_by_pid_map and renamed accessing functions in accordance to comments below.

Reply:
sorry, I know I suggested it, but `pkey_by_pid_map` might not be great, since you're actually trying to track thread IDs. I made that comment before I saw that it was tracking tids.  i would suggest using `thread_id` instead of `tid`. `pid` is kind of ubiquitous, but `tid` can be confusing.

Another good option here is to define a typename for the mapping you want that is descriptive, and then the name of the map is less important, since its type has a sane name. 

this is up to you, but at minimum `pid` shouldn't be in the name, since you're actually tracking this per thread.

Reply:
Added a `thread_id` type name for the type of the map and then changed `pid` to `tid` for better clarity.